src/main/java/com/adyen/httpclient/ClientInterface.java

```diff
@@ -25,8 +25,10 @@
 import com.adyen.Config;
 
 public interface ClientInterface {
+    
```

Comment:
"post" is used for DL and doesn't set any authentication headers. Why do we need to introduce postWithApiKeyFlag?

To be backwards compatible we can keep "request" method as one that uses basic auth and just introduce requestWithApiKey or smth for the one using x-api-key

Reply:
Thanks @lancergr 

Removed API authentication for post.

About request method backward compatibility, I believe this method is not externally exposed. This is part of internal client (HttpURLConnectionClient) used to send request to core system and request method is always called from Resource class. Please share your views more on this.

Reply:
I checked more in documentations for this.  I found "Proxy configuration" in readme file and it is making object of HttpURLConnectionClient class. I think your point is good to keep old request method, just in case someone ever uses it in external code. Updated code for the same. thanks.